drive/src/drive/document/insert.rs

```diff
@@ -477,14 +491,19 @@ impl Drive {
                     .clone()
                     .add_path_info(index_path_info.clone());
 
-                // here we are inserting an empty tree that will have a subtree of all other index properties
-                self.batch_insert_empty_tree_if_not_exists(
-                    path_key_info,
-                    &storage_flags,
-                    apply,
-                    transaction,
-                    &mut batch_operations,
-                )?;
+                if !path_key_info.is_contained_in_cache(&batch_insertion_cache) {
```

Comment:
PathKeyInfo doesn't know about cache anything, I wouldn't add cache-related methods there. Let's create a structure for cache so it will have those methods and accept path key info as a param.

Reply:
The problem is that a cache won't contain path key info's either, so this was the best compromise that I thought made sense.